### PR TITLE
feat: allow to the endpoint url to boto3

### PR DESCRIPTION
**Pull request by pmdroid**
## Summary
This allow to use S3 compatible storage like minio or Scaleway.

## Changes
We added the optional parameter endpoint_url for creating S3 storage.

## Importance
This allows any S3 compatible object storage to be used (not just AWS).

## Checklist

This PR:

- [ ] adds new tests (if appropriate)
- [ ] adds a change file in the `changes/` directory (if appropriate)
- [X] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

**Review by desertaxle — CHANGES_REQUESTED** [review]
Thanks for your contribution! This is a very useful addition! I had a suggestion for the documentation, but feel free to refine further if you think it's necessary.

**Comment by pmdroid** [comment]
@desertaxle Hey thanks for checking, unfortunately i cannot see your suggested changes :(

File: docs/concepts/storage.md [hunk 1]
```diff
@@ -72,4 +72,9 @@ Updated default storage!
...
 To see the IDs of available storage configurations, use the `prefect storage ls` command.
 
-`prefect storage reset-default` resets the default storage option to temporary local storage.
\ No newline at end of file
+`prefect storage reset-default` resets the default storage option to temporary local storage.
+
+### S3 compatible storage
+When need there is the option to use a S3 compatible storage like minio or Scaleway.
+To get this working you would need to provide the endpoint url when creating the storage via the CLI.
+Usually the endpoint url looks something like this ```https://bucketname.s3.nl-ams.scw.cloud```.
```
Comment (desertaxle) [inline docs/concepts/storage.md:80]:
```suggestion
### S3 Endpoint URL
When using an S3 compatible storage solution, like minio or Scaleway, or a VPC endpoint for S3, specify the endpoint URL when setting up storage via the CLI. An endpoint URL is not necessary when using AWS S3 directly.
```

**Comment by desertaxle** [comment]
> unfortunately i cannot see your suggested changes :(
I guess it didn't go through before. You should be able to see it now!

**Comment by pmdroid** [comment]
@desertaxle thanks for the changes  👍

**Comment by xyxz-web** [comment]
would be great if this could be implemented soon, e.g. the only non-local storage I can access in my company is a custom (internal) S3 endpoint.

**Comment by xyxz-web** [comment]
> May I suggest to also add a key_prefix to avoid storage at the "root" of a bucket. This involves two changes in storage.py in addition to adding the function arguments:
> 
> s3_client.upload_fileobj(Fileobj=stream, Bucket=self.bucket, Key=self.key_prefix +"/" + key) s3_client.download_fileobj(Bucket=self.bucket, Key=self.key_prefix +"/" + key, Fileobj=stream)

This is a great suggestion, but is this related to the endpoint url in any way?

**Comment by madkinsz** [comment]
Hi! While this is still relevant for the `prefect-aws` package, we're removing the `S3Storage` object from the core library very soon.

I'd suggest using the `RemoteFileSystem` instead https://orion-docs.prefect.io/concepts/filesystems/#remote-file-system

We're including examples for using MinIO in the next release.

**Comment by madkinsz** [comment]
I believe this has been resolved.

**Comment by AyrtonDare** [comment]
Hi @zanieb do you have any MinIO examples? 

prefect-aws always seems to refer to S3 regardless of the endpoint_url provided and there seems to be no way to use minio in the PrefectHQ docs

**Pull request closed without merging**